gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -130,4 +134,16 @@ private static List<ProxyBackendConfiguration> upcast(List<GatewayBackend> gatew
         }
         return proxyBackendConfigurations;
     }
+
+    private List<GatewayBackend> fetchAllBackends()
+    {
+        try {
+            List<GatewayBackend> backends = gatewayBackendSupplier.get();
+            cache.set(backends);
```

Comment:
Trying to understand the logic here since we are kind of layering two caches on top of each other -- the memoized supplier and the `AtomicReference`. I guess you are making it such that:

- We use the memoized supplier to reduce DB load / increase performance for frequent re-lookups (within 500ms)
- We use the AtomicReference, which never expires, to provide a _stale_ view of the data when the DB is inaccessible

This makes sense to me, but it's a bit confusing to track how these two interact. Why not just use a Guava Cache? It gives you much more configurability/flexibility so we can have all of this logic in a single place (using `refreshAfterWrite()`, which does exactly what we want here) and gives us more hooks to add monitoring:
```java
    private static final Logger LOG = Logger.get(HaGatewayManager.class);
    private static final Object ALL_BACKEND_CACHE_KEY = new Object();

    private final GatewayBackendDao dao;
    private final LoadingCache<Object, List<GatewayBackend>> backendCache;

    private final CounterStat backendLookupSuccesses = new CounterStat();
    private final CounterStat backendLookupFailures = new CounterStat();


    public HaGatewayManager(Jdbi jdbi)
    {
        dao = requireNonNull(jdbi, "jdbi is null").onDemand(GatewayBackendDao.class);
        backendCache = CacheBuilder
                .newBuilder()
                .initialCapacity(1)
                .refreshAfterWrite(Duration.ofSeconds(1))
                .build(CacheLoader.asyncReloading(
                        CacheLoader.from(this::loadAllBackends),
                        MoreExecutors.listeningDecorator(Executors.newSingleThreadExecutor())));
    }

    private List<GatewayBackend> loadAllBackends()
    {
        try {
            var backends = dao.findAll();
            backendLookupSuccesses.update(1);
            return backends;
        } catch (Exception e) {
            backendLookupFailures.update(1);
            LOG.warn(e, "Failed to load backends");
            throw e;
        }
    }

    // probably should be named "getAllBackends()" since it may or may not actually do any fetch
    private List<GatewayBackend> fetchAllBackends()
    {
        return backendCache.getUnchecked(ALL_BACKEND_CACHE_KEY);
    }
```